realm/realm-library/src/main/java/io/realm/DynamicRealm.java

```diff
@@ -241,42 +220,14 @@ public void executeTransaction(Transaction transaction) {
         return RealmResults.createFromDynamicTableOrView(this, tableView, className);
     }
 
-    private static DynamicRealm create(RealmConfiguration configuration) {
-        synchronized (BaseRealm.class) {
-
-            // Check if a cached instance already exists for this thread
-            String canonicalPath = configuration.getPath();
-            Map<RealmConfiguration, Integer> localRefCount = referenceCount.get();
-            Integer references = localRefCount.get(configuration);
-            if (references == null) {
-                references = 0;
-            }
-            Map<RealmConfiguration, DynamicRealm> realms = realmsCache.get();
-            DynamicRealm realm = realms.get(configuration);
-            if (realm != null) {
-                localRefCount.put(configuration, references + 1);
-                return realm;
-            }
-
-            // Create new DynamicRealm and cache it. All exception code paths must close the DynamicRealm otherwise we risk
-            // serving faulty cache data.
-            validateAgainstExistingConfigurations(configuration);
-            boolean autoRefresh = Looper.myLooper() != null;
-            realm = new DynamicRealm(configuration, autoRefresh);
-            List<RealmConfiguration> pathConfigurationCache = globalPathConfigurationCache.get(canonicalPath);
-            if (pathConfigurationCache == null) {
-                pathConfigurationCache = new CopyOnWriteArrayList<RealmConfiguration>();
-                globalPathConfigurationCache.put(canonicalPath, pathConfigurationCache);
-            }
-            pathConfigurationCache.add(configuration);
-            realms.put(configuration, realm);
-            localRefCount.put(configuration, references + 1);
-
-            // Increment global reference counter
-            realm.acquireFileReference(configuration);
-
-            return realm;
-        }
+    /**
+     * Creates a {@link DynamicRealm} instance without checking the existence in the {@link RealmCache}.
+     *
+     * @return a {@link DynamicRealm} instance.
+     */
+    static DynamicRealm createInstance(RealmConfiguration configuration) {
```

Comment:
Ok, just I would just keep it as it is then. We definitely want to avoid having locks outside RealmCache. I would probably describe the use case a bit better in the Javadoc though as it sounds like this should only be called from RealmCache?